Logging/.CSProject/Logger/OneTimeLogger.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Anvil.CSharp.Logging
+{
+    /// <summary>
+    /// A wrapper for <see cref="Logger"/> that only allows a message to be emitted from a given call site once per session.
+    /// A call site is determined to be unique by a combination of its file path and line number.
+    /// </summary>
+    public readonly struct OneTimeLogger : ILogger
+    {
+        private static readonly HashSet<int> s_CalledLogSites = new HashSet<int>();
+
+        /// <summary>
+        /// Resets <see cref="OneTimeLogger"/>'s to allow one time logs to be emitted one more time.
+        /// </summary>
+        public static void Reset()
+        {
+            s_CalledLogSites.Clear();
+        }
+
+
+        private readonly Logger m_Logger;
+
+
+        internal OneTimeLogger(in Logger logger)
```

Comment:
Are the double newlines intentional?

Reply:
Yea, I like to double new line between inner class blocks, static blocks, field/property blocks, and method blocks.

If it really bugs someone we can have a little style fight 🤷 . It's not a big deal in Rider but I find it helps readability in editors that don't insert annotations above declaration lines.